crypto/s2n_drbg.c

```diff
@@ -61,10 +61,10 @@ static int s2n_drbg_bits(struct s2n_drbg *drbg, struct s2n_blob *out)
 
 static int s2n_drbg_update(struct s2n_drbg *drbg, struct s2n_blob *provided_data)
 {
-    uint8_t temp[32];
-    struct s2n_blob temp_blob = {.data = temp,.size = sizeof(temp) };
+    uint8_t temp[S2N_MAX_SEED_SIZE];
+    struct s2n_blob temp_blob = {.data = temp,.size = s2n_drbg_seed_size(drgb) };
```

Comment:
I have that check in s2n_drbg_instantiate, do you think it needs to be added to the other methods that use this pattern?

Reply:
This seems like a relatively useful construct, so I agree we should be a standard way to do it. Since it results in a blob being defined on the stack, I think it needs to be a macro. I haven't tested the following but it should be a good starting point.

```
#define S2N_STACK_BLOB (name, size, maximum) \
	size_t name ## _requested_size = (size); \
	size_t name ## _max_size = (maximum); \
	uint8_t name ## _buf[name ## _requested_size] = {0}; \
	lte_check(name ## _requested_size, name ## _max_size}; \
	struct s2n_blob name = {.data = name ## _buf, .size = name ## _requested_size)
```
Then, the start of this method can simply be replaced with `S2N_STACK_BLOB(temp_blob, s2n_drbg_seed_size(drgb), S2N_MAX_SEED_SIZE);` (Code hasn't been tested personally.)

Reply:
Went with a macro inspired by Salusa, but I have to use the maximum directly otherwise the compiler can't figure out its a constant.